components/core/src/clp/TransactionManager.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef CLP_TRANSACTIONMANAGER_HPP
+#define CLP_TRANSACTIONMANAGER_HPP
+
+#include <type_traits>
+
+namespace clp {
+/**
+ * Class to perform different actions depending on whether a transaction succeeds or fails. The
+ * default state assumes the transaction fails.
+ * @tparam SuccessHandler A cleanup lambda to call on success.
+ * @tparam FailureHandler A cleanup lambda to call on failure.
+ */
+template <typename SuccessHandler, typename FailureHandler>
+requires(std::is_invocable_v<SuccessHandler> && std::is_invocable_v<FailureHandler>)
```

Comment:
Based on a suggestion from the Rabbit, can we require that the lambdas are `noexcept`? If not, we should document that they should be.